### DISCUSSION TITLE
Search with vector embeddings

### DISCUSSION THREAD

**Discussion by DominikGrodl** [firebase/firebase-ios-sdk#12942] [General]
Hello,

are there currently any plans to include vector search also in mobile client SDKs? I see they are already available for Python and Node and was wondering whether there is a plan to bring them to mobile clients, and if so, what is the ETA on that. We currently have a use-case for vector similarity search and were debating using Firestore as a backing storage to facilitate that.

Thanks and have a great day!

**Comment by peterfriese** [comment]
Hi Dominik,

Thanks for reaching out! Supporting other client SDKs is something that we're looking into, and we'd be interested in hearing more about your use case (if you can share).

In the meantime, you can use the [Vector Search with Firestore extension](https://extensions.dev/extensions/googlecloud/firestore-vector-search) - it supports computing embeddings, and provides a callable function that you can use to perform semantic search queries.

Cheers,
Peter

**Reply by DominikGrodl** [reply]
Thanks for the Swift reply, Peter! The use case is very simple, what I am looking for is just a vector similarity search and sorting capability. Similar to how I would use [pgvector](https://github.com/pgvector/pgvector) when working with Postgres databases, i.e. I can pass in a vector and filter/sort data based on that.
